content/en/docs/contribute/_index.md

```diff
@@ -71,6 +71,16 @@ documentation, but it should help you get started.
   - Propose improvements to docs tests
   - Propose improvements to the Kubernetes website or other tooling
 
+## Contributions best practices
+
+- Do write clear and meaningful GIT commit messages.
+- Make sure to include _Github Special Keywords_ which references the issue and automatically closes the issue when PR is merged.
+- When you make a small change to a PR like fixing a typo, any style change, or changing grammar. Make sure you squash your commits so that you dont get a large number of commits for a relatively small change.
+- Make sure you include a nice PR description depicting the code you have changes, why to change a following piece of code and ensuring there is sufficient information for the reviewer to understand your PR.
+- Additional Readings : 
+    - [chris.beams.io/posts/git-commit/](https://chris.beams.io/posts/git-commit/)
+    - [github.com/blog/1506-closing-issues-via-pull-requests ](https://github.com/blog/1506-closing-issues-via-pull-requests )
+    - [davidwalsh.name/squash-commits-git ](https://davidwalsh.name/squash-commits-git )
```

Comment:
@vishakhanihore how about adding this to https://kubernetes.io/docs/contribute/start/#improve-existing-content and / or https://kubernetes.io/docs/contribute/intermediate/ and then signposting readers there?

There is already a set of instructions on how to squash commits, for example.

(also, Prow auto-squashes commits for this repo if I'm remembering correctly)

Reply:
@sftim Thanks for the suggestions, I will quickly look into it 👍